adaptor-rm-webapp/src/main/java/com/sample/rm/RMToolManager.java

```diff
@@ -50,6 +53,8 @@
 
     // Start of user code class_attributes
 	private static final Logger log = LoggerFactory.getLogger(RMToolManager.class);
+    private static Map<String, Requirement> requirements = new HashMap<String, Requirement>(1000);
```

Comment:
well, that's not production-ready. moreover, it's not easy to replace this, e.g. by supplying another impl of an interface

Reply:
Of course not. Sample adaptor is not meant to be production-ready.